server/src/fhir/cacheStructureDefinitions.ts

```diff
@@ -0,0 +1,72 @@
+const nodeCache = require('node-cache')
+const _ = require('lodash')
+
+const cache = new nodeCache()
+
+export const setStructureDefinition = (fhirStructureDefinition: any) => {
+  const struct = buildCustomStructure(fhirStructureDefinition.resource)
+
+  // Use id as key. If it isn't present, use url
+  const key = struct.id || struct.url
+
+  if (key === undefined) {
+    throw new Error('Structure definition has no id nor url field.')
+  }
+
+  // Put obj in key value store
+  cache.set(key, struct)
+}
+
+export const getStructureDefinition = (key: string) => cache.get(key)
+
+const buildCustomStructure = (structResource: any): any => {
+  if (!('snapshot' in structResource)) {
+    throw new Error('Snapshot is needed in the structure definition.')
+  }
+
+  // Create the new custom structure
+  var customStruct = {}
+
+  Object.keys(structResource).forEach(key => {
+    if (structureFieldsWhiteList.indexOf(key) === -1) return
+    _.set(customStruct, key, structResource[key])
+  })
+
+  // If the structure defines a primitive type (one which we don't need to unroll in UI)
+  // we don't need the properties field
+  if (structResource.kind !== 'primitive-type') {
```

Comment:
what happens if the definition is a `primitive-type` indeed ? it's not even in the final structure right now

Reply:
If it's a primitive type, we already have basic information in structure as name, description, ...
I don't think we need something more in Pyrog, do we?